src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -199,31 +232,218 @@ private Future<String> fetchToken(JsonObject payload, String tenant,
-          response.statusCode()));
-      }
-      logger.debug("Got token " + token + " from authz");
-      return token;
+
+      return response.bodyAsJsonObject();
     });
   }
 
-  private Future<String> fetchRefreshToken(String userId, String sub, String tenant,
-    String okapiURL, String requestToken) {
-    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + "/refreshtoken");
+  private Future<HttpResponse<Buffer>> logout(String tenant,
+      String okapiURL, String accessToken, String refreshToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_LOGOUT_ENDPOINT);
+
     request.putHeader(XOkapiHeaders.TENANT, tenant)
-      .putHeader(XOkapiHeaders.TOKEN, requestToken);
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
 
-    JsonObject payload = new JsonObject().put("userId", userId).put("sub", sub);
-    return request
-      .expect(ResponsePredicate.SC_CREATED)
-      .expect(ResponsePredicate.JSON)
-      .sendJsonObject(payload)
-      .map(response -> response.bodyAsJsonObject().getString("refreshToken")
-      );
+      return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private Future<HttpResponse<Buffer>> logoutAll(String tenant,
+      String okapiURL, String accessToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_LOGOUT_ALL_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.send();
+  }
+
+  private Future<HttpResponse<Buffer>> fetchRefreshToken(String tenant,
+      String okapiURL, String accessToken, String refreshToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_REFRESH_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private boolean usesTokenSignLegacy(String endpoint) {
+    return endpoint.equals(TOKEN_SIGN_ENDPOINT_LEGACY);
+  }
+
+
+  // For documentation of the cookie attributes please see:
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+  private String refreshTokenCookie(String refreshToken, String okapiUrl, String refreshTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(refreshTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    var domain = URI.create(okapiUrl).getHost();
+    return Cookie.cookie(REFRESH_TOKEN, refreshToken)
+        .setMaxAge(ttlSeconds)
+        .setSecure(true)
+        .setPath("/authn")
+        .setHttpOnly(true)
+        .setDomain(domain)
+        .setSameSite(CookieSameSite.NONE)
```

Comment:
Why is NONE needed?
Can we use the most secure STRICT?

Reply:
Ok this one is tricky. The values don't really line up semantically in my mind with how they are implemented. But my understanding from reading around the web is that NONE is needed for CORS to work -- which we definitely need in our case. This is a relatively new thing that changed with a breaking change to Chrome about a year and a half ago.

This isn't documentation but it gives you a flavor of what I'm talking about: https://stackoverflow.com/questions/46288437/set-cookies-for-cross-origin-requests. LMK what you think.

Reply:
And also I think strict won't work for CORS.